src/main/java/org/datavyu/plugins/ffmpeg/FfmpegMediaPlayer.java

```diff
@@ -57,4 +71,68 @@ protected synchronized void playerSetMute(boolean enable) throws MediaException
             }
         }
     }
+
+    @Override
+    protected synchronized void playerUpdateCurrentTime() {
+        if (isUpdateTimeEnabled
+                || Double.compare(playerGetPresentationTime(), streamPrevTime) != 0 ) {
+            double presentationTime = playerGetPresentationTime();
+            if (presentationTime >= 0.0) {
+                double newTime = presentationTime;
+                if (Double.compare(newTime, streamPrevTime) != 0) {
+                    if(Math.abs(newTime - masterCurrentTime) >= SYNC_THRESHOLD){
+                        seek(masterCurrentTime);
+                        streamCurrentTime = masterCurrentTime;
+                    } else {
+                        streamCurrentTime = newTime;
+                    }
+                    streamPrevTime = newTime;
+                }
+            }
+        }
+    }
+
+    protected void createMediaTimer() {
+        synchronized (MediaTimerTask.timerLock) {
+            if (mediaTimerTask == null) {
+                mediaTimerTask = new MediaTimerTask(this);
+                mediaTimerTask.start();
+            }
+        }
+    }
+
+    protected void destroyMediaTimer() {
+        synchronized (MediaTimerTask.timerLock) {
+            if (mediaTimerTask != null) {
+                isUpdateTimeEnabled = false;
+                mediaTimerTask.stop();
+                mediaTimerTask = null;
+            }
+        }
+    }
+
+    @Override
+    public void updateMasterTime() {
+        double masterClockTime = (double) masterClock.getTimeUpdate(this);
+        if(masterClockTime != 0.0){
```

Comment:
I will change this condition to masterClockTime != masterCurrentTime since we are updating the Clock Timer in Datavyu each 100 ms, we would update only when DV CLock and FFmpeg plugin clock are different